### DISCUSSION TITLE
Replacement for  dTE.Events.DTEEvents.OnStartupComplete

### DISCUSSION THREAD

**Discussion by scottcantwell** [VsixCommunity/Community.VisualStudio.Toolkit#395] [Q&A] [answered]
For some reason, this is event is no longer being hooked up. Is there a replacement for the Startup Event for VS in the Toolkit?

```
protected override async Task InitializeAsync(CancellationToken cancellationToken, IProgress<ServiceProgressData> progress)
{

var serviceContainer = (IServiceContainer)this;

dTE = (DTE2) serviceContainer.GetService(typeof(DTE));

dTE.Events.DTEEvents.OnStartupComplete += OnStartupComplete;

}
```

**Comment by reduckted** [comment, marked as answer]
It depends on what you're trying to achieve. Extensions are loaded asynchronously, so it's very likely that the IDE has finished loading by the time your extension starts loading.

What were you doing in that event handler that couldn't be done sooner?

**Reply by scottcantwell** [reply]
I am hooking up more events in the Startup event handler, but I can move those out and handle them much earlier.
